modules/core/05-port/types/module.go

```diff
@@ -108,6 +108,28 @@ type IBCModule interface {
 	) error
 }
 
+// IBCModuleV2 defines an interface that implements all the callbacks
+// that modules must define as specified in IBC Protocol V2
+type IBCModuleV2 interface {
```

Comment:
not convinced this is the best place for this interface, open to suggestions! (Also do we want V2 in the name? )

Reply:
It really should be referenced as `api.IBCModule`, I wonder if it makes sense to create the api directory now, would be a weird partial transition if you didn't move over the light client module

Main dependency concern is that if you move this api elsewhere, you may need to define interfaces or move the concrete implementations